cdap-ui/app/cdap/components/KeyValuePairs/KeyValueStore.js

```diff
@@ -70,7 +71,7 @@ const keyValues = (state = initialState, action = defaultAction) => {
       }
       return stateCopy;
     case KeyValueStoreActions.onReset:
-      return [];
+      return initialState;
```

Comment:
I think you might have to recreate the `initialState` in here. I think the reducer might have modified the actual value of pairs inside initialState